urlead5353366_Zeldon_BigData_Class_tgz-pJ8-LocalTestsJ8/src/com/beust/jcommander/defaultprovider/PropertyFileDefaultProvider.java

```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright (C) 2010 the original author or authors.
+ * See the notice.md file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.beust.jcommander.defaultprovider;
+
+import com.beust.jcommander.IDefaultProvider;
+import com.beust.jcommander.ParameterException;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.Properties;
+
+/**
+ * A default provider that reads its default values from a property file.
+ * 
+ * @author cbeust
+ */
+public class PropertyFileDefaultProvider implements IDefaultProvider {
+  public static final String DEFAULT_FILE_NAME = "jcommander.properties";
+  private Properties m_properties;
+
+  public PropertyFileDefaultProvider() {
+    init(DEFAULT_FILE_NAME);
+  }
+
+  public PropertyFileDefaultProvider(String fileName) {
+    init(fileName);
+  }
+
+  private void init(String fileName) {
+    try {
+      m_properties = new Properties();
+      URL url = ClassLoader.getSystemResource(fileName);
+      if (url != null) {
+        m_properties.load(url.openStream());
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the resource returned by the following method call: *openStream*. Execution paths that do not contain closure statements were detected. Ensure the object returned by *openStream()* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).